gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/DuplicateObservationStatementMapper.java

```diff
@@ -66,37 +64,51 @@ private static List<String> getLinksetsIn(RCMRMT030101UKEhrComposition ehrCompos
                 .toList();
     }
 
-    private static void findAndMergeNonTruncatedObservationStatementIntoTruncatedObservationStatement(
-            RCMRMT030101UKObservationStatement truncatedObservationStatement, List<RCMRMT030101UKComponent4> components) {
+    private static void findAndMergeNonTruncatedObservationStatementIntoTruncatedObservationStatementOrRemoveDuplicates(
+            RCMRMT030101UKObservationStatement observationStatementWithPertinentInfo, List<RCMRMT030101UKComponent4> components) {
```

Comment:
Similar to the comment above, this method is quite large in size - are we able to break it down at all? 😄

Reply:
ok, good suggestion, I will try to make it smaller